packages/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx

```diff
@@ -0,0 +1,152 @@
+import {
+  AbstractWidgetProps,
+  StagePanelLocation,
+  StagePanelSection,
+  StageUsage,
+  UiItemsProvider,
+} from "@bentley/ui-abstract";
+import {
+  SpatialContainmentTree,
+  ClassGroupingOption,
+  UiFramework
+} from "@bentley/ui-framework";
+import React from "react";
+import { TreeWidgetComponent } from "./TreeWidgetComponent";
+import { CategoriesTreeComponent } from "./trees/CategoriesTree";
+import { ModelsTreeComponent } from "./trees/ModelsTree";
+import { IModelConnection, Viewport } from "@bentley/imodeljs-frontend";
+import { SelectableContentDefinition } from "@bentley/ui-components";
+import { TreeWidget } from "../TreeWidget";
+
+export interface TreeWidgetUiItemsProviderProps {
```

Comment:
Redefining everything in TreeWidgetControlOptions seems unnecessary for when all you're trying to do is making the iModelConnection optional.

I __think__ this will suffice, please test it out

```ts
export interface TreeWidgetUiItemsProviderProps extends TreeWidgetControlOptions {
   iModelConnection?: IModelConnection;
}
```

Reply:
I tried doing this before making the new interface and i got an error, that's why i made this one

Reply:
What was the error?

Reply:
Interface 'TreeWidgetUiItemsProviderProps' incorrectly extends interface 'TreeWidgetControlOptions'.
  Types of property 'iModelConnection' are incompatible.
    Type 'IModelConnection | undefined' is not assignable to type 'IModelConnection'.
      Type 'undefined' is not assignable to type 'IModelConnection'.

Reply:
Ok guess you can only extend and make an optional prop required, not vice versa :/

Anyway, can type the arg in the constructor as `Partial<TreeWidgetControlOptions>` like so: 
![image](https://user-images.githubusercontent.com/11051042/125337811-6d2aa480-e31d-11eb-8596-10adc30ecd7d.png)